dali/base/dautils.cpp

```diff
@@ -1949,16 +1960,22 @@ class CTimedCache
 
     unsigned check()
     {
+        unsigned expired = 0;
```

Comment:
Could do with some comments explaining what code is doing.

/* items are ordered, such that oldest items are at the start.
    This method scans through from oldest to newest until the
    current item's "due" time has not expired.
    It then removes all up to that point, i.e. those that have expired,
    and returns the timing difference between now and the next due time. */